### PR TITLE
`Development`: Prefer Optional.orElseThrow() over get() when no presence checks are used

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context
The modernizer plugin currently complains about every usage of `Optional.get()`. As discussed in the developer meeting, using `get()` is fine if a `isPresent()` check got performed instead. If we do not perform this check and want an exception to be thrown if the value is missing, `orElseThrow()` should be used.

### Description
This PR applies the IntelliJ "Optional.get() is called without isPresent() check" and replaces the found issues with `orElseThrow()`.

The modernizer still complains about more `get()` usages since it considers the whole method as legacy API. I added a exclusion rule to suppress the warning.
grafik

### Steps for Testing
code review.

This PR simply applied search and replace. The only file with additional changes was `TutorialGroupIntegrationTest` due to its heavliy duplicated use of `userRepository.findOneByLogin`.

If you scroll through the changes and find other places for code improvements, place note them down in this PR. However, I'd like to keep this as a search and replace PR and implement these suggestions in a followup.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### Screenshots
grafik

grafik

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportService.java [hunk 1]
```diff
@@ -199,7 +199,7 @@ private void cloneAndEnableAllBuildPlans(ProgrammingExercise templateExercise, P
         final var templateKey = templateExercise.getProjectKey();
         final var targetKey = newExercise.getProjectKey();
         final var targetName = newExercise.getCourseViaExerciseGroupOrCourseMember().getShortName().toUpperCase() + " " + newExercise.getTitle();
-        continuousIntegrationService.get().createProjectForExercise(newExercise);
+        continuousIntegrationService.orElseThrow().createProjectForExercise(newExercise);
         continuousIntegrationService.get().copyBuildPlan(templateKey, templatePlanName, targetKey, targetName, templatePlanName, false);
         continuousIntegrationService.get().copyBuildPlan(templateKey, solutionPlanName, targetKey, targetName, solutionPlanName, true);
         continuousIntegrationService.get().givePlanPermissions(newExercise, templatePlanName);
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportService.java:205]: This is one example of bad code style as described in the PR description. To keep this PR simple, I'll refactor this in a Followup-PR.

**Review by simon-hng — DISMISSED** [review]
Great addition! I always thought that this was strange and my IDE was complaining the entire time

**Review by tobias-lippert — DISMISSED** [review]
Code LGTM

**Review by laadvo — APPROVED** [review]
reapprove after merge

**Review by bassner — DISMISSED** [review]

**Review by bassner — APPROVED** [review]

**Pull request merged**